fuzz/fuzz_targets/snapshot2.rs

```diff
@@ -0,0 +1,150 @@
+        };
+        let offset = std::cmp::min(offset as usize, data.len());
+        let full_length = data.len() - offset;
+        let size = if length > 0 {
+            std::cmp::min(full_length, length as usize)
+        } else {
+            full_length
+        };
+        Ok((data.slice(offset..offset + size), full_length as u64))
+    }
+}
+
+fn build_machine() -> DefaultMachine<Box<AsmCoreMachine>> {
+    let isa = ISA_IMC | ISA_A | ISA_B | ISA_MOP;
+    let core_machine = AsmCoreMachine::new(isa, VERSION2, u64::max_value());
+    DefaultMachineBuilder::new(core_machine).build()
+}
+
+fuzz_target!(|data: [u8; 32]| {
+    let mut deque = Deque::new(data);
+    let dummy_data = {
+        let mut program = vec![0u8; deque.u16() as usize + 1024];
+        for i in 0..program.len() {
+            program[i] = (i % 256) as u8;
+        }
+        let mut content = vec![0u8; deque.u16() as usize + 1024];
+        for i in 0..content.len() {
+            content[i] = (i / 256) as u8;
+        }
+        DummyData {
+            program: program.into(),
+            content: content.into(),
+        }
+    };
+    let mut loading_action_vec: Vec<LoadingAction> = Vec::new();
+    for i in 0..2 + deque.u8() as usize % 3 {
+        let segsize = deque.u16() as u64 % 1024;
+        let p_vaddr = i as u64 * 1024 * 256;
+        let p_memsz = segsize;
+        let p_offset = deque.u32() as u64 % (dummy_data.program.len() as u64 - segsize);
+        let p_filesz = segsize;
+        let aligned_start = round_page_down(p_vaddr);
+        let padding_start = (p_vaddr).wrapping_sub(aligned_start);
+        let size = round_page_up((p_memsz).wrapping_add(padding_start));
+        let slice_start = p_offset;
+        let slice_end = p_offset.wrapping_add(p_filesz);
+        assert!(slice_start <= slice_end);
+        loading_action_vec.push(LoadingAction {
+            addr: aligned_start,
+            size,
+            flags: FLAG_EXECUTABLE | FLAG_FREEZED,
+            source: slice_start as u64..slice_end as u64,
+            offset_from_addr: padding_start,
+        })
+    }
+    let mut ctx = Snapshot2Context::new(dummy_data.clone());
+    let metadata = ProgramMetadata {
+        actions: loading_action_vec,
+        entry: 0,
+    };
+
+    let mut machine1 = build_machine();
+    let mut machine2 = build_machine();
+    machine1
+        .load_program_with_metadata(&dummy_data.program, &metadata, &vec![])
+        .unwrap();
+    ctx.mark_program(&mut machine1, &metadata, &0, 0).unwrap();
+    for i in 0..2 + deque.u8() as usize % 3 {
+        let id = i as u32 % 2;
+        let data = match id {
+            DATA_SOURCE_PROGRAM => &dummy_data.program,
+            DATA_SOURCE_CONTENT => &dummy_data.content,
+            _ => unreachable!(),
+        };
+        let length = deque.u16() as u64 % 1024;
+        let offset = deque.u32() as u64 % (data.len() as u64 - length);
+        let addr = 1024 * 1024 + (deque.u8() as u64 % 4) * 1024 * 256;
```

Comment:
The `addr` can be anything. We can ignore some errors from store_bytes functions.